libraries/core_libs/consensus/include/pbft/block_proposer.hpp

```diff
@@ -46,13 +46,16 @@ class SortitionPropose : public ProposeModelFace {
       : dag_mgr_(std::move(dag_mgr)), dag_blk_mgr_(std::move(dag_blk_mgr)), trx_mgr_(std::move(trx_mgr)) {
     LOG_OBJECTS_CREATE("PR_MDL");
     LOG(log_nf_) << "Set sortition DAG block proposal" << dag_blk_mgr_->sortitionParamsManager().getSortitionParams();
+    // Add a random component in proposing stale blocks so that not all nodes propose stale blocks at the same time
+    // This will make stale block be proposed after waiting random interval between 2 and 20 seconds
+    max_num_tries_ += (node_addr_[0] % (10 * max_num_tries_));
```

Comment:
what is this code? do we want it in develop?

Reply:
Yes, this is something I did on testnet and I think we should keep on develop. It is adding a random component to creating stale blocks so that multiple nodes do not propose stale blocks at exactly same time. Probably there is a better way to implement this and not use max_num_tries_ for the delay but I would keep this for the merge and maybe improve on this later